### DISCUSSION TITLE
Hot Module Replacement

### DISCUSSION THREAD

**Discussion by yoniholmes** [react-figma/react-figma#412] [Q&A]
First of all, thank you for the plugin, it's great!

I'm having difficulty getting HMR to work. I've followed the instructions at https://react-figma.dev/docs/hmr.

With the plugin running in my terminal (via `webpack && webpack-dev-server --config webpack.ui.config.js`), and with the Figma console open, I can see when I make changes to `App.tsx` that the Figma console reports 'Updated modules', correctly identifying that my source code has changed. However, it also says 'App is up to date', and my changes are not reflected in Figma. 

I notice that my `dist` doesn't appear to update, and my `manifest.json` differs from: https://github.com/react-figma/react-figma/blob/master/examples/basic-hmr/manifest.json. Mine is based on https://github.com/react-figma/react-figma-boilerplate/blob/master/manifest.json, the difference being mine says `"ui": "dist/ui.html"` whereas the HMR example has: `"ui": "ui.html"`. No idea if this is relevant. But the docs didn't say I should change anything. 

I also notice that the boilerplate codebase diverges quite a lot from the 'Basic HMR' codebase, so it's proving difficult to work out how to combine the right parts from each repo to get this working.

Does anyone have any ideas, or have you had any similar problems getting HMR to work?

Any advice much appreciated!

Many thanks! 🙏 

Jonathan

**Comment by DaleWebb** [comment]
I'm a little late to the party... but I had issues with HMR not working either. After a few hours, I found that the `dist/ui.html` file had the `webpack`-ed bundle inside it, along with the:
```html
<script type="text/javascript" src="http://localhost:8080/ui.js"></script>
```

When I deleted `webpack`-ed bundled code, and only left the `<script />`, the HMR appeared to work.

I hope that helps someone...
